osgp/platform/osgp-adapter-domain-core/src/main/java/org/opensmartgridplatform/adapter/domain/core/application/tasks/DeviceMessageCleanupJob.java

```diff
@@ -0,0 +1,160 @@
+
+    @Value("${osgp.scheduling.job.database.cleanup.device.message.retention}")
+    private int deviceMessageRetentionPeriodInMonths;
+
+    @Value("${osgp.scheduling.job.database.cleanup.device.message.csv.file.location}")
+    private String csvFileLocation;
+
+    @Value("${osgp.scheduling.job.database.cleanup.device.message.csv.file.prefix}")
+    private String csvFilePrefix;
+
+    @Value("${osgp.scheduling.job.database.cleanup.device.message.csv.file.compression.enabled}")
+    private boolean csvFileCompressionEnabled;
+
+    @Value("${osgp.scheduling.job.database.cleanup.device.message.page.size}")
+    private int deviceMessagePageSize;
+
+    @Autowired
+    private TransactionalDeviceLogItemService transactionalDeviceLogItemService;
+
+    @Override
+    public void execute(final JobExecutionContext context) throws JobExecutionException {
+        LOGGER.info("Quartz triggered cleanup of database - device message records.");
+        final DateTime start = DateTime.now();
+
+        final Date retention = this.calculateDate();
+        final List<DeviceLogItem> oldDeviceMessages = this.transactionalDeviceLogItemService
+                .findDeviceLogItemsBeforeDate(retention, this.deviceMessagePageSize);
+        if (!oldDeviceMessages.isEmpty()) {
+            this.saveDeviceMessagesToCsvFile(oldDeviceMessages);
+
+            LOGGER.info("Deleting device messages...");
+            this.transactionalDeviceLogItemService.deleteDeviceLogItems(oldDeviceMessages);
+
+            System.gc();
+        }
+
+        final DateTime end = DateTime.now();
+        LOGGER.info("Start: {}, end: {}, duration: {} milliseconds.", start, end, end.getMillis() - start.getMillis());
+    }
+
+    private Date calculateDate() {
+        final Date date = DateTime.now().minusMonths(this.deviceMessageRetentionPeriodInMonths).toDate();
+        LOGGER.info("Determined date: {} based on device message retention period in months: {}.", date,
+                this.deviceMessageRetentionPeriodInMonths);
+
+        return date;
+    }
+
+    private void saveDeviceMessagesToCsvFile(final List<DeviceLogItem> deviceMessages) {
+        try {
+            LOGGER.info("Converting device messages ...");
+            final String[] header = DeviceMessageToStringArrayConverter.getDeviceMessageFieldNames();
+            final List<String[]> lines = DeviceMessageToStringArrayConverter.convertDeviceMessages(deviceMessages);
+            LOGGER.info("Device messages converted.");
+
+            final String csvFilePath = this.writeCsvFile(this.csvFileLocation, this.csvFilePrefix, header, lines);
+
+            if (this.csvFileCompressionEnabled) {
+                this.compressCsvFile(csvFilePath);
+            }
+        } catch (final Exception e) {
+            LOGGER.error("Exception during CSV file creation, compression or device message deletion.", e);
+        }
+    }
+
+    private final static class DeviceMessageToStringArrayConverter {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a private constructor to hide the implicit public one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1118)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reorder the modifiers to comply with the Java Language Specification. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AModifiersOrderCheck)